query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +44,37 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Duration timeSeriesPeriod;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesPeriod = setTimeSeriesPeriod(request);
     analyze(request);
   }
 
+  private Duration setTimeSeriesPeriod(QueryRequest request) {
```

Comment:
```suggestion
  private Duration calculateTimeSeriesPeriod(QueryRequest request) {
```

Setters return void and change state, this produces a result which is set externally.